src/stream/product.rs

```diff
@@ -1,5 +1,6 @@
-use core::pin::Pin;
+use alloc::boxed::Box;
```

Comment:
`Box` should already be available as a global, shouldn't it?

Reply:
Oh I guess not in `core`. Makes sense then (: 

https://doc.rust-lang.org/core/prelude/v1/index.html